File: domain/src/test/java/no/nav/melosys/domain/behandlingsgrunnlag/SoeknadTest.java
package no.nav.melosys.domain.behandlingsgrunnlag;

import java.util.Collections;
import java.util.Set;

import no.nav.melosys.domain.behandlingsgrunnlag.data.SelvstendigForetak;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class SoeknadTest {
    @Test
    public void hentAlleOrganisasjonsnumre() {
        SelvstendigForetak selvstendigForetak = new SelvstendigForetak();
        selvstendigForetak.orgnr = "12345678910";

        Soeknad soeknad = new Soeknad();
        soeknad.selvstendigArbeid.selvstendigForetak = Collections.singletonList(selvstendigForetak);

        String orgNr2 = "10987654321";
        soeknad.juridiskArbeidsgiverNorge.ekstraArbeidsgivere.add("10987654321");

        Set<String> organisasjonsnumre = soeknad.hentAlleOrganisasjonsnumre();
        assertThat(organisasjonsnumre.size()).isEqualTo(2);
        assertThat(organisasjonsnumre).contains(selvstendigForetak.orgnr)
            .contains(orgNr2);
    }
}

File: domain/src/test/kotlin/no/nav/melosys/domain/mottatteopplysninger/SoeknadTest.kt
package no.nav.melosys.domain.mottatteopplysninger

import io.kotest.matchers.collections.shouldContain
import io.kotest.matchers.shouldBe
import no.nav.melosys.domain.mottatteopplysninger.data.SelvstendigForetak
import org.junit.jupiter.api.Test

internal class SoeknadTest {
    @Test
    fun hentAlleOrganisasjonsnumre() {
        val selvstendigForetak = SelvstendigForetak().apply {
            orgnr = "12345678910"
        }
        val soeknad = Soeknad()
        soeknad.selvstendigArbeid.selvstendigForetak = listOf(selvstendigForetak)

        val orgNr2 = "10987654321"
        (soeknad.juridiskArbeidsgiverNorge.ekstraArbeidsgivere as MutableList).add(orgNr2)


        val organisasjonsnumre = soeknad.hentAlleOrganisasjonsnumre()


        organisasjonsnumre.run {
            size shouldBe 2
            shouldContain(selvstendigForetak.orgnr)
            shouldContain(orgNr2)
        }
    }
}